### DISCUSSION TITLE
Make life easier for SABnzbd-on-Docker users: CLI parameters for Wizard, and tell if Complete nicely exposed to Host

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2720] [Ideas]
Two things that might be useful for SAB-on-docker users, both related to "exposed directory" ... if that is the word for the directory that's visible on the host.

I'm not a docker expert: I only use SAB-on-docker for small tests.

**Two Wizard CLI parameters**

LSIO says: : "In Sabnzbd gui settings, under Folders, make sure to set the Completed Download Folder as /downloads and the Temporary Download Folder as /incomplete-downloads". https://docs.linuxserver.io/images/docker-sabnzbd/#application-setup:~:text=In%20Sabnzbd%20gui%20settings%2C%20under%20Folders

Manual work for a user. So a source of problems.

Suggestion: create two extra CLI options to specify the Temp and Complete ... but only if the wizard is run ie on first run

`--WizardIncompleteDir=/incomplete-downloads --WizardCompleteDir=/downloads`

The LSIO people could then use those parameters in https://github.com/linuxserver/docker-sabnzbd/blob/437c449e4a663dddd816345eafafe82d41bb0457/root/etc/s6-overlay/s6-rc.d/svc-sabnzbd/run#L16

If we are willing & able to do this, before we do, we should ask the LSIO people if they see it as an advantage

**If in docker, tell if Complete is nicely exposed, and warn it's relevant name outside the docker**

It's needed that the Complete is exposed to the Host. Otherwise, the downloaded stuff is not usable.
Plus: that inside-docker directory name is not relevant outside the docker, like Sonarr etc.

So SAB, if  in docker, tell if Complete is nicely exposed, and warn it's relevant name outside the docker

```
sander@zwart2204:~$ sudo docker run --net=host -v /home/sander/docker-spul:/outsidestuff  -it ubuntu:latest /bin/bash

Inside the container:
root@zwart2204:/# mount | grep -i outside
/dev/nvme0n1p2 on /outsidestuff type ext4 (rw,relatime,errors=remount-ro)
```

Note: AFAIK, inside the docker you cannot find out the directory name outside the docker. So SAB cannot tell that.

**Comment by Safihre** [comment]
Wouldn't it be easier if they just provide a basic sabnzbd.ini when the file doesn't exists, with just these 2 values? That's what we also do for SynoCommunity.

**Reply by sanderjo** [reply]
But then you dont get the wizard?

**Reply by Safihre** [reply]
We show the wizard if no servers are defined, unrelated to the folders.

**Comment by thezoggy** [comment]
Setting the paths via cli or env vars would deff would be useful for many.

I will note that LSIO default docker paths are generally not recommended from *arr community and such as it's usually an indication of badly setup paths because of crossing mounts which means no atomic moves. 

They are 'simple' for people to get started that have no clue, even lsio themselves mention how its not optimal:
https://hub.docker.com/r/linuxserver/sabnzbd#Download%20folders

the normal recommendation is to use trash's guide for setup:
https://trash-guides.info/Downloaders/SABnzbd/Paths-and-Categories/
